include/pybind11/pybind11.h

```diff
@@ -2163,25 +2166,29 @@ template <class T> function get_overload(const T *this_ptr, const char *name) {
     .. code-block:: cpp
 
       std::string toString() override {
-        PYBIND11_OVERLOAD_NAME(
+        PYBIND11_OVERRIDE_NAME(
             std::string, // Return type (ret_type)
             Animal,      // Parent class (cname)
             "__str__",   // Name of method in Python (name)
             toString,    // Name of function in C++ (fn)
         );
       }
 \endrst */
-#define PYBIND11_OVERLOAD_NAME(ret_type, cname, name, fn, ...) \
-    PYBIND11_OVERLOAD_INT(PYBIND11_TYPE(ret_type), PYBIND11_TYPE(cname), name, __VA_ARGS__) \
-    return cname::fn(__VA_ARGS__)
+#define PYBIND11_OVERRIDE_NAME(ret_type, cname, name, fn, ...) \
+    do { \
+        PYBIND11_OVERRIDE_IMPL(PYBIND11_TYPE(ret_type), PYBIND11_TYPE(cname), name, __VA_ARGS__); \
+        return cname::fn(__VA_ARGS__); \
+    } while (false)
```

Comment:
> Is that a good change, the `do { ... } while (false)`?

Assuming we want to make this kind of a breaking change, sure.

Reply:
Is that ever going to break something that wasn't broken before, is the main question?
I can only imagine `if (...) PYBIND11_OVERRIDE(...);` to actually get fixed, rather than broken?

Reply:
Oh, yes. I still need to fix that, such that the original macros are the same as they were. That was the whole point of this exercise!